src/NuGet.Clients/PackageManagement.VisualStudio/SolutionRestore/SolutionRestoreWorker.cs

```diff
@@ -53,7 +59,28 @@ internal sealed class SolutionRestoreWorker : ISolutionRestoreWorker, IDisposabl
             _serviceProvider = serviceProvider;
 
             _componentModel = _serviceProvider.GetService<SComponentModel, IComponentModel>();
+#if VS15
+            _solutionManager = _componentModel.GetService<IVsSolutionManager>();
 
+            if (_solutionManager == null)
+            {
+                throw new ArgumentNullException(nameof(_solutionManager));
+            }
+
+            IVsSolution vsSolution = serviceProvider.GetService(typeof(SVsSolution)) as IVsSolution;
+            if (vsSolution != null)
```

Comment:
Exactly. The product code ought to be clean of test related assumptions. In this particular case no test exists that demands this condition.